views/collection.py

```diff
@@ -18,7 +18,29 @@ def __init__(self) -> None:
     async def callback(self, interaction: discord.Interaction) -> None:
         await interaction.response.defer()
         await self.view.send_msg(1)
+
+class GalleryBtn(discord.ui.Button):
+    def __init__(self) -> None:
+        self.view: CollectionView
+        super().__init__(emoji="🎨", label="Send To Gallery")
+        self.last_send_time: float = 0
+
+    async def callback(self, interaction: discord.Interaction) -> None:
+        await interaction.response.defer()
+        if all(item is None for item in self.view.cards):
+            return await interaction.followup.send("You can't send this collection without any cards in it.", ephemeral=True)
         
+        if time.time() < (self.last_send_time + 30):
+            return await interaction.followup.send("Whoa, that was too fast! Could you please try again later?", ephemeral=True)
+        self.last_send_time = time.time()
+
+        image_bytes, image_format = await asyncio.to_thread(iufi.gen_cards_view, self.view.cards, size_rate=1)
+        if self.view.cards and (gallery_channel := interaction.guild.get_channel(func.settings.GALLERY_CHANNEL_ID)):
+            await gallery_channel.send(
+                content=f"Check out the collection `{self.view.sel_collection.title()}` curated by {interaction.user.mention}!",
```

Comment:
now i gotta name my collection properly 😢. If i want a custom caption, then I'll have to post manually right? or is there anything we can do here?

Reply:
Oh we add a modal to let player to type their own captions